lib/config.go

```diff
@@ -155,6 +158,16 @@ type RuntimeConfig struct {
 	// Hooks List of hooks to run with container
 	Hooks map[string]HookParams
 
+	// PIDNamespace selects the PID namespace scope.  Choose from
+	// 'container' for all containers (including pod infra containers)
+	// to have sibling PID namespaces (the default), 'pod' for all
+	// containers to share a single, per-pod namespace, or
+	// 'pod-container' to have the pod's infra container in one PID
+	// namespace with the non-infra containers in per-container PID
+	// namespaces that are children of the pod's infra PID namespace .
+	// A 'hostPID' Kubernetes pod specification overrides this setting.
+	PIDNamespace string `toml:"pid_namespace"`  // FIXME: should this be an enum type?
```

Comment:
We don't want options for all possible pid namespace sharing options at the daemon level.
We only want to be able to distinguish whether we want pod or pod-container when k8s passes us POD as the sharing option:
```
enum NamespaceMode {
    POD       = 0;
    CONTAINER = 1;
    NODE      = 2;
    TARGET    = 3;
 }
```

Reply:
The reason we had the option at the daemon level was because pid namespace support wasn't fully baked in k8s and we wanted a control at the daemon level. With the upcoming changes, we can do what k8s requests through the API and choice will be down to what to do when we get `POD` as the setting.

Reply:
> The reason we had the option at the daemon level was because pid namespace support wasn't fully baked in k8s…

kubernetes/community#1048 should help with this, but I'm not clear on the deployment procedure.  Wouldn't we need changes to kubernetes/api to implement 
kubernetes/community#1048 before we could rely on them here?  And until we have k8s API support for kubernetes/community#1048, I'm not sure how to adjust this.  Should we just table this PR until there is k8s API support for kubernetes/community#1048?

And either way, I think maintainers here are going to have to make a call on strings vs. enums for the CRI-O config.  The kubernetes/community#1048 support question is just impacting the option name and scope, we'll still need an option that takes string arguments on the command line, and holds that configuration in Go.

Reply:
> kubernetes/community#1048 should help with this, but I'm not clear on the deployment procedure. Wouldn't we need changes to kubernetes/api to implement
kubernetes/community#1048 before we could rely on them here? And until we have k8s API support for kubernetes/community#1048, I'm not sure how to adjust this. Should we just table this PR until there is k8s API support for kubernetes/community#1048?

Yes, we can wait for the implementation of 1048 to land: https://github.com/kubernetes/kubernetes/pull/58716 

> And either way, I think maintainers here are going to have to make a call on strings vs. enums for the CRI-O config. The kubernetes/community#1048 support question is just impacting the option name and scope, we'll still need an option that takes string arguments on the command line, and holds that configuration in Go.

We can use the approach that we have for `ImageVolumes` where we have defined a custom type.